Packs/CortexXDR/Integrations/PaloAltoNetworks_XDR/PaloAltoNetworks_XDR.py

```diff
@@ -1693,6 +1693,29 @@ def endpoint_scan_command(client, args):
     )
 
 
+def sort_by_key(list_to_sort, main_key, fallback_key):
+    """Sorts a given list elements by main_key for all elements with the key,
+    uses sorting by fallback_key on all elements that dont have the main_key"""
+    list_elements_with_main_key = [element for element in list_to_sort if element.get(main_key)]
+    sorted_list = sorted(list_elements_with_main_key, key=itemgetter(main_key))
+    if len(list_to_sort) == len(sorted_list):
+        return sorted_list
+
+    list_elements_without_main_key = [element for element in list_to_sort if not element.get(main_key)]
+    list_elements_with_fallback_key = [element for element in list_elements_without_main_key
```

Comment:
It could, it is handling the alert list of an incident - that could be quite long. why?

Reply:
because if it is very long we might need to reconsider what @teizenman suggested below, to be more efficient. We can make it one list comprehension instead of one

Reply:
I see your point though I think the current flow is more readable, I'll make some adjustments.

Reply:
I agree, it's readability VS efficiency.
Your call 100%